pkg/crc/api/handlers.go

```diff
@@ -0,0 +1,102 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/machine"
+	"github.com/code-ready/crc/pkg/crc/validation"
+	"github.com/code-ready/crc/pkg/crc/version"
+)
+
+func statusHandler(_ ArgsType) string {
+	statusConfig := machine.ClusterStatusConfig{Name: constants.DefaultName}
+	clusterStatus, err := machine.Status(statusConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	return encodeStructToJson(clusterStatus)
+}
+
+func stopHandler(_ ArgsType) string {
+	stopConfig := machine.StopConfig{
+		Name:  constants.DefaultName,
+		Debug: true,
+	}
+	commandResult, err := machine.Stop(stopConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	return encodeStructToJson(commandResult)
+}
+
+func startHandler(_ ArgsType) string {
+	startConfig := machine.StartConfig{
+		Name:          constants.DefaultName,
+		BundlePath:    crcConfig.GetString(config.Bundle.Name),
+		VMDriver:      crcConfig.GetString(config.VMDriver.Name),
+		Memory:        crcConfig.GetInt(config.Memory.Name),
+		CPUs:          crcConfig.GetInt(config.CPUs.Name),
+		NameServer:    crcConfig.GetString(config.NameServer.Name),
+		GetPullSecret: getPullSecretFileContent,
+		Debug:         true,
+	}
+	status, err := machine.Start(startConfig)
+	if err != nil {
+		logging.Error(err.Error())
+		return "Failed"
+	}
+	return encodeStructToJson(status)
+}
+
+func versionHandler(_ ArgsType) string {
+	v := &machine.VersionResult{
+		CrcVersion:       fmt.Sprintf("%s+%s", version.GetCRCVersion(), version.GetCommitSha()),
```

Comment:
Well, no particular reason, i wanted to replicate the same information shown as part of the `crc version`. Yes sure if you want i can add the helper `machine.GetVersion()` and call that here. I didn't do that because i don't think it'd be used anywhere else, for `crc version` we just compose these information in the `cmd` package itself.

Reply:
We should at least drop the `fmt.Sprintf` and split it across 2 fields in the `VersionResult` struct.

Reply:
Okay, will do that!